### DISCUSSION TITLE
exporting objects as types "mesh, object3D" and PIV

### DISCUSSION THREAD

**Discussion by wileyknight** [iimachines/Maya2glTF#161] [Q&A]
I was wondering why my exports have a mixture of being exported as type: object3D and type: mesh with and without a name change to the geometry name PIV (ex. myObjectPIV)? Sometimes a single object is exported with both nodes for a object3d and a mesh but only the PIV version has the attached geometry and is of type mesh. Other times it will export fine without a PIV change on the name.

All of my objects in the scene are polygonal meshes with no history and frozen translations, rotations, and scales. Some meshes are combined but that doesn't seem to be a factor.

**Comment by ziriax** [comment]
Maya2glTF can insert extra glTF nodes in order te reconstruct the Maya DAG. Compared to Maya, glTF is very limited. The PIV nodes are needed to simulate the Maya pivot point. Only the rotation pivot point is supported, not the scaling one, the would require even more nodes, and is rarely needed.

Maya2glTF also creates extra nodes if you have joints with segment scale compensation enabled (the default in Maya unfortunately).

You can try Maya's "Bake Pivot" command to try get rid of the pivot nodes. Basically every Maya object that has a non zero pivot point will get two glTF nodes.
